test/utils/noderesourcetopologies/noderesourcetopologies.go

```diff
@@ -66,6 +67,29 @@ func GetUpdated(cli client.Client, ref nrtv1alpha2.NodeResourceTopologyList, tim
 	return updatedNrtList, err
 }
 
+func GetUpdatedForNode(cli client.Client, ctx context.Context, ref nrtv1alpha2.NodeResourceTopology, timeout time.Duration) (nrtv1alpha2.NodeResourceTopology, error) {
+	var equalZones bool
+	var updatedNrt nrtv1alpha2.NodeResourceTopology
+	klog.Infof("NRT change: reference is %s", e2enrt.ToString(ref))
+	err := wait.PollImmediate(5*time.Second, timeout, func() (bool, error) {
+		err := cli.Get(ctx, client.ObjectKeyFromObject(&ref), &updatedNrt)
+		if err != nil {
+			klog.Errorf("cannot get the updated NRT object %s/%s", ref.Namespace, ref.Name)
+			return false, err
+		}
+		// very cheap test to rule out false negatives
+		if updatedNrt.ObjectMeta.ResourceVersion == ref.ObjectMeta.ResourceVersion {
+			klog.Warningf("NRT for %q resource version didn't change", ref.Name)
+			return false, nil
+		}
+		equalZones = apiequality.Semantic.DeepEqual(ref.Zones, updatedNrt.Zones)
+		klog.Infof("NRT change: updated to %s", e2enrt.ToString(updatedNrt))
+		return !equalZones, nil
```

Comment:
IIUC the function will return with the first NRT it spots. Yet, it won't be enough because releasing/consuming resources cannot be controlled, it periodically getting updated. What do you think about using the ForNodeResourceTopologiesSettled before taking the final decision and return with the updated NRT?

Reply:
yes, it will return once it detects a change (any change) in NRT counters. We need indeed to have stronger guarantees, for example waiting for NRT to settle before look for the value. We probably need to iterate more in this area